File: Java/problem1111.java
class Solution {

    public int[] maxDepthAfterSplit(String seq) {
        char[] str = seq.toCharArray();
        int strLen = str.length;
        int stackLen = 0;
        int[] result = new int[strLen];
        for (int i = 0; i < strLen; ++i) {
            if (str[i] == '(') {
                result[i] = ++stackLen % 2;
            }
            else {
                result[i] = stackLen-- % 2;
            }
        }
        return result;
    }
}

File: python/problem1111.py
class Solution:
    def maxDepthAfterSplit(self, seq: str) -> List[int]:
        result = []
        stackLen = 0
        for s in seq:
            if s == '(':
                stackLen += 1
                result.append(stackLen % 2)
            else:
                result.append(stackLen % 2)
                stackLen -= 1
        return result
